colossus/src/main/scala/colossus/controller/InputController.scala

```diff
@@ -206,5 +214,10 @@ trait InputController[Input, Output] extends MasterController[Input, Output] {
 
 
   protected def processMessage(message: Input)
+  
+  protected def processBadRequest(reason: Throwable): Output = {
+    disconnect()
+    throw reason
```

Comment:
So there's a few issues here.  

1.  it shouldn't throw the exception, it should just disconnect.  
2.  There need to be two new methods.  `processBadRequest`needs to be abstract, so that we force subclasses to implement it.  Then we need another method, let's call it `fatalInputError` that does 3 things: it calls `processBadRequest`, it writes the `Output`object that it returns, and it calls `disconnect()`.  It has to work this way because when an error like this occurs, we have no option but to close the connection.  So therefore it should not be up to subclasses to call disconnect, and we don't want that behavior changed if they override the method.
2.  The methods need to be defined in `Controller`, not `InputController`, because it needs to be able to push the output object, which `InputController`cannot do.